### PR TITLE
DPC 971 Feature: Add Snyk scans for Github Actions

### PR DESCRIPTION
**Pull request by lostpurplecow**
### Fixes [DPC-971](https://jira.cms.gov/browse/DPC-971)

Currently, we are performing Snyk analysis after a branch gets merged to master (triggered by Jenkins). This is great, but it means that a build could fail if a vulnerability was introduced in a feature branch (it isn't detectable by Snyk until the code makes it to master).

We should add a Snyk analysis into our feature branch CI via a Github Action.

https://github.com/snyk/actions

### Proposed Changes

- Add `snyk/actions` to Github Actions

### Change Details

- Fail on high severity failures
- Add SNYK_API_TOKEN to Github Repo Secrets and KeyBase

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [ ] no PHI/PII is affected by this change

### Acceptance Validation

Can only test Workflow change via a PR...

- https://github.com/CMSgov/dpc-app/actions/runs/432800008

Will block success of action run.

- Match this: https://app.snyk.io/org/oeda/reports/

### CODE REVIEW

**Review by msnook — CHANGES_REQUESTED** [review]
I'm curious why there are Snyk blocking failures in this feature branch, but the scans occurring from `master` (via Jenkins) are clean.  At the moment, this PR is blocked because of the vulnerabilities detected via Snyk.  Are there configuration differences between the Snyk integration in Actions vs the Snyk integration in Jenkins?

**Comment by lostpurplecow** [comment]
> I'm curious why there are Snyk blocking failures in this feature branch, but the scans occurring from `master` (via Jenkins) are clean. At the moment, this PR is blocked because of the vulnerabilities detected via Snyk. Are there configuration differences between the Snyk integration in Actions vs the Snyk integration in Jenkins?

The configuration for this GA is the same as the one in Jenkins as you can see here:
```groovy    
stage('Snyk Test') {
      steps {
        dir('dpc-app') {
          snykSecurity failOnIssues: !params.SNYK_OVERRIDE, severity: 'high', snykInstallation: 'dpc-snyk', snykTokenId: 'SNYK_API_TOKEN'
        }
      }
    }
```

**Comment by msnook** [comment]
> > I'm curious why there are Snyk blocking failures in this feature branch, but the scans occurring from `master` (via Jenkins) are clean. At the moment, this PR is blocked because of the vulnerabilities detected via Snyk. Are there configuration differences between the Snyk integration in Actions vs the Snyk integration in Jenkins?
> 
> The configuration for this GA is the same as the one in Jenkins as you can see here:
> 
> ```groovy
> stage('Snyk Test') {
>       steps {
>         dir('dpc-app') {
>           snykSecurity failOnIssues: !params.SNYK_OVERRIDE, severity: 'high', snykInstallation: 'dpc-snyk', snykTokenId: 'SNYK_API_TOKEN'
>         }
>       }
>     }
> ```
> 
> Hmm though judging from this snapshot of last successful build there were no issues:
> https://app.snyk.io/org/oeda/project/70afbed4-f7f9-45ae-8309-f70a09ed2541/history/7a5093ef-c3b9-4f15-a1b6-71ac9ec5bc8a
> 
> Actually, w/ GA we'd have to add the file paths for targetting dependencies... w/ Jenkins it is only going for root... therefore it may seem we have discovered a configuration issue o.O

Cool, is the configuration issue in Jenkins (meaning that there are vulnerabilities we need to fix now)?  Might we need to use `targetFile` in the Jenkins config, similar to how we're using `--file` in the GA config?

**Comment by lostpurplecow** [comment]
> > > I'm curious why there are Snyk blocking failures in this feature branch, but the scans occurring from `master` (via Jenkins) are clean. At the moment, this PR is blocked because of the vulnerabilities detected via Snyk. Are there configuration differences between the Snyk integration in Actions vs the Snyk integration in Jenkins?
> > 
> > 
> > The configuration for this GA is the same as the one in Jenkins as you can see here:
> > ```groovy
> > stage('Snyk Test') {
> >       steps {
> >         dir('dpc-app') {
> >           snykSecurity failOnIssues: !params.SNYK_OVERRIDE, severity: 'high', snykInstallation: 'dpc-snyk', snykTokenId: 'SNYK_API_TOKEN'
> >         }
> >       }
> >     }
> > ```
> > 
> > 
> > Hmm though judging from this snapshot of last successful build there were no issues:
> > https://app.snyk.io/org/oeda/project/70afbed4-f7f9-45ae-8309-f70a09ed2541/history/7a5093ef-c3b9-4f15-a1b6-71ac9ec5bc8a
> > Actually, w/ GA we'd have to add the file paths for targetting dependencies... w/ Jenkins it is only going for root... therefore it may seem we have discovered a configuration issue o.O
> 
> Cool, is the configuration issue in Jenkins (meaning that there are vulnerabilities we need to fix now)? Might we need to use `targetFile` in the Jenkins config, similar to how we're using `--file` in the GA config?

Yes, that means there are vulnerabilities we'd need to fix now... 
Addressing here: https://github.com/CMSgov/dpc-ops/pull/282

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -14,6 +15,12 @@ jobs:
     steps:
       - name: "Checkout code"
         uses: actions/checkout@v1
+      - name: Run Snyk - Java
+        uses: snyk/actions/maven-3-jdk-11@master
+        continue-on-error: false
+        with:
+          command: test
+          args: --severity-threshold=high --file=pom.xml
```
Comment (dhgreene) [inline .github/workflows/ci-workflow.yml:23]: Should we include the other sources here, or perhaps use the directory-driven approach from the ops PR?

**Review by orenfromberg — APPROVED** [review]
tentatively approving, don't merge yet.

**Comment by orenfromberg** [comment]
:taco:

**Pull request merged**